pkg/runtime/fake.go

```diff
@@ -147,6 +149,58 @@ type FakeRuntime struct {
 	FakeExecutor *FakeExecutor
 	FakeProfiler *FakeProfiler
 	FakeLock     *FakeLock
+	FakeSignal   *FakeSignal
+}
+
+// FakeSignal implements a fake signal handling for testing
+type FakeSignal struct {
+	lock    sync.Mutex
+	signals map[os.Signal]bool
+	channel chan os.Signal
+}
+
+// NewFakeSignal returns a FakeSignal
+func NewFakeSignal() *FakeSignal {
+	return &FakeSignal{
+		channel: make(chan os.Signal),
+		signals: map[os.Signal]bool{},
+	}
+}
+
+// Notify implements Signal's interface Notify method
+func (f *FakeSignal) Notify(signals ...os.Signal) <-chan os.Signal {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	for _, s := range signals {
+		f.signals[s] = true
+	}
+	return f.channel
+}
+
+// Reset implements Signal's interface Reset method
+func (f *FakeSignal) Reset(signals ...os.Signal) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	if len(signals) == 0 {
+		f.signals = map[os.Signal]bool{}
+	} else {
+		for _, s := range signals {
+			delete(f.signals, s)
+		}
+	}
+}
+
+// SendSignal sends the given signal to the signal notification channel if the signal was
+// previously specified in a call to Notify
+func (f *FakeSignal) SendSignal(signal os.Signal) {
```

Comment:
The `SendSignal` name seems a bit redundant given it receives an argument that the caller knows is a signal. Do you think it would make sense to name it just `Send`?